### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by alan-wu**
Use the latest flatmapvuer which resolves bug on minimap and unintended graphics on maps.

Also fix a bug causing markers to not displayed properly occasionally.

### CODE REVIEW

File: src/components/viewers/MultiFlatmap.vue [hunk 1]
```diff
@@ -188,10 +217,10 @@ export default {
           this.toggleSyncMode();
       }
     },
-    multiFlatmapReady: function (component) {
+    multiFlatmapReady: function () {
       this.$refs.multiflatmap.getCurrentFlatmap().enablePanZoomEvents(true); // Use zoom events for dynamic markers
-      this.updateMarkers(component);
       this.flatmapReady = true;
+      this.flatmapMarkerZoomUpdate(true);
       //component.addPanZoomEvent();
```
Comment (agarny) [inline src/components/viewers/MultiFlatmap.vue:224]: Not part of this PR, I know, but can't say that I like code that is commented out. It's like at lines 268-271 in fact. After all, that's why we have versioning.

File: src/components/viewers/MultiFlatmap.vue [hunk 2]
```diff
@@ -24,6 +24,35 @@ import EventBus from "../EventBus";
 import store from "../../store";
 import markerZoomLevels from '../markerZoomLevels';
 
+/*
+ * Function to check markers visibility at zoom level.
+ * I have modified it to make sure the marker is displayed
+ * if the uberon is not present in the hardcoded zoom-level list.
+ */
```
Comment (agarny) [inline src/components/viewers/MultiFlatmap.vue:31]: Might want to rephrase:
```
Check markers' visibility at the given zoom level.
```
The rest of the comment is already captured in this PR (although I would have expected a GitHub issue for this PR).

File: src/components/viewers/MultiFlatmap.vue [hunk 3]
```diff
@@ -24,6 +24,35 @@ import EventBus from "../EventBus";
...
+
+const checkMarkersAtZoomLevel = (flatmapImp, markers, zoomLevel) => {
+  if (markers) {
+    markers.forEach(id=>{
+      let foundInArray = false;
+      //First check if uberon is in the list, check for zoom level 
+      //if true. Note: markerZoomLevels is imported.
```
Comment (agarny) [inline src/components/viewers/MultiFlatmap.vue:38]: Space between `//` and the text.

File: src/components/viewers/MultiFlatmap.vue [hunk 4]
```diff
@@ -24,6 +24,35 @@ import EventBus from "../EventBus";
...
 import markerZoomLevels from '../markerZoomLevels';
 
+/*
+ * Function to check markers visibility at zoom level.
+ * I have modified it to make sure the marker is displayed
+ * if the uberon is not present in the hardcoded zoom-level list.
+ */
+
+const checkMarkersAtZoomLevel = (flatmapImp, markers, zoomLevel) => {
+  if (markers) {
```
Comment (agarny) [inline src/components/viewers/MultiFlatmap.vue:34]: This method is only used here and replaces some code where `markers` was always defined, so... probably no need to test `markers` here?

File: src/components/viewers/MultiFlatmap.vue [hunk 5]
```diff
@@ -24,6 +24,35 @@ import EventBus from "../EventBus";
...
+    markers.forEach(id=>{
```
Comment (agarny) [inline src/components/viewers/MultiFlatmap.vue:35]: Space before and after `=>`?

File: src/components/viewers/MultiFlatmap.vue [hunk 6]
```diff
@@ -24,6 +24,35 @@ import EventBus from "../EventBus";
...
+      let foundInArray = false;
+      //First check if uberon is in the list, check for zoom level 
+      //if true. Note: markerZoomLevels is imported.
+      for (let i = 0; i < markerZoomLevels.length; i++) {
```
Comment (agarny) [inline src/components/viewers/MultiFlatmap.vue:39]: I personally prefer `++i` to `i++` since it usually results in slightly (ok, negligibly) faster code.

File: src/components/viewers/MultiFlatmap.vue [hunk 7]
```diff
@@ -206,14 +235,8 @@ export default {
...
-        let markers = store.state.settings.markers
-        markers.forEach(id=>{
-          markerZoomLevels.map(el=>{
-            if (el.id === id && this.zoomLevel >= el.showAtZoom) {
-              flatmapImp.addMarker(id, "simulation");
-            }
-          })
-        })
+        let markers = store.state.settings.markers;
+        checkMarkersAtZoomLevel(flatmapImp, markers, this.zoomLevel);
```
Comment (agarny) [inline src/components/viewers/MultiFlatmap.vue:239]: Based on the current code in `checkMarkersAtZoomLevel()`, it looks like this line could simply be replaced with:
```javascript
markers.forEach(id => {
  flatmapImp.addMarker(id, "simulation");
});
```

Comment (alan-wu) [reply src/components/viewers/MultiFlatmap.vue:239]: No, it does not. checkMarersAtZoomLevel does not add any marker that fails the condition.

File: src/components/viewers/MultiFlatmap.vue (hunk 5 above)
Comment (alan-wu) [reply src/components/viewers/MultiFlatmap.vue:35]: Sure.

File: src/components/viewers/MultiFlatmap.vue (hunk 4 above)
Comment (alan-wu) [reply src/components/viewers/MultiFlatmap.vue:34]: The previous code does not check markers. I do not want to make the assumption that markers is defined.

File: src/components/viewers/MultiFlatmap.vue (hunk 7 above)
Comment (agarny) [reply src/components/viewers/MultiFlatmap.vue:239]: Oops, my bad, I focussed on `foundInArray` and overlooked your `zoomLevel >= markerZoomLevels[i].showAtZoom` test!

File: src/components/viewers/MultiFlatmap.vue (hunk 4 above)
Comment (agarny) [reply src/components/viewers/MultiFlatmap.vue:34]: Was just trying to be consistent, but agreed.

**Review by agarny — APPROVED** [review]
